app/modules/key_management/secret_manager.py

```
@@ -189,3 +204,68 @@
             return {"message": "Secret deleted successfully"}
         except Exception as e:
             raise HTTPException(status_code=404, detail=f"Secret not found: {str(e)}")
+
+    @router.post("/api-keys", response_model=APIKeyResponse)
+    async def create_api_key(
+        user=Depends(AuthService.check_auth),
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)